packages/components/src/DataList/DataList.types.ts

```diff
@@ -289,6 +289,11 @@ export interface DataListActionProps<T extends DataListObject> {
    */
   readonly destructive?: boolean;
 
+  /**
+   * Determine if the action is visible for a given item.
+   */
+  readonly visible?: (item: T) => boolean;
```

Comment:
What would are thoughts be on allowing the row action to not be visible for specific rows? I think it would be nice but I think there may be some UX considerations.

Reply:
There are a lot of factors in hiding an action on a single item/row. In Jobber's sense, we have statuses. You can archive pretty much any state except for something that is already archived. So you would want an archive action for anything that isn't in an archived state.

Reply:
I was mainly thinking about hiding [this](https://github.com/GetJobber/atlantis/blob/f07dac102920d9a611f43dbc89c340dacc2c00fc/packages/components/src/DataList/DataList.types.ts#L232)
So it doesn't have the cursor if the onClick wouldn't do anything.

As I think more I don't really think it has value with what we need the DataList to do right now. I think this is something to consider in the future _if_ teams ask for it and it makes sense

Reply:
OH, like the whole thing. With the amount of actions we're thinking of doing, I don't think we need that yet

Reply:
Agreed